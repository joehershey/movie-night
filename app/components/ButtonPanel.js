import React from "react";
import {
  StyleSheet,
  View,
  Text,
  Image,
  TouchableWithoutFeedback,
} from "react-native";

import { COLORS, STYLES } from "../assets/saved";
import { FontAwesome5 } from "@expo/vector-icons";

function ButtonPanel(props) {
  const showCreateGroup = props.showCreateGroup;
  const showJoinGroup = props.showJoinGroup;
  const toggleShowCreateGroup = props.toggleShowCreateGroup;
  const toggleShowJoinGroup = props.toggleShowJoinGroup;
  return (
    <View style={{ flexDirection: "row", marginBottom: 30 }}>
      <TouchableWithoutFeedback
        onPress={() => toggleShowCreateGroup(!showCreateGroup)}
      >
        <View style={[STYLES.createGroupButton, STYLES.btn]}>
          <Text style={[{ color: "white", fontSize: 20 }]}>Create Group</Text>
        </View>
      </TouchableWithoutFeedback>
      <TouchableWithoutFeedback
        onPress={() => toggleShowJoinGroup(!showJoinGroup)}
      >
        <View style={[STYLES.joinGroupButton, STYLES.btn]}>
          <Text style={[{ color: COLORS.primary, fontSize: 20 }]}>
            Join Group
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
  );
}

export default ButtonPanel;
