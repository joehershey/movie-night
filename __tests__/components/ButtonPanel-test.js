import ButtonPanel from "../../app/components/ButtonPanel";
import React from 'react';
import renderer from 'react-test-renderer';

describe('Test Button Panel', () => {
    beforeEach( async () => {
        global.tree = renderer.create(<ButtonPanel />);
    });

    it("Button panel renders properly", async () => {
        await expect(global.tree).toMatchSnapshot();
    });
});