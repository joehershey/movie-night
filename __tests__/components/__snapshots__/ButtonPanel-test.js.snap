// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Panel Button panel renders properly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 30,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#166fdb",
          "height": 70,
          "top": 10,
          "width": "40%",
        },
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "justifyContent": "center",
          "margin": 5,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
          },
        ]
      }
    >
      Create Group
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "height": 70,
          "top": 10,
          "width": "40%",
        },
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "justifyContent": "center",
          "margin": 5,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1d1f1e",
            "fontSize": 20,
          },
        ]
      }
    >
      Join Group
    </Text>
  </View>
</View>
`;
